src/tests/tribol_proximity_check.cpp

```diff
@@ -0,0 +1,274 @@
+                                        contact_surf_2, tribol::SURFACE_TO_SURFACE, tribol::NO_CASE,
+                                        tribol::COMMON_PLANE, tribol::FRICTIONLESS, tribol::PENALTY, binning_method );
+    tribol::setMfemKinematicConstantPenalty( coupling_scheme_id, penalty, penalty );
+    tribol::setBinningProximityScale( coupling_scheme_id, binning_proximity );
+
+    UpdateTribol( mesh, coupling_scheme_id );
+  }
+
+  void SetUp2DCommonPlaneProblem( tribol::BinningMethod binning_method )
+  {
+    constexpr int dim = 2;
+    // polynomial order of the finite element discretization
+    auto order = std::get<0>( GetParam() );
+    // binning proximity value
+    auto binning_proximity = std::get<1>( GetParam() );
+    // mesh interpenetration
+    auto mesh_interpenetration = std::get<2>( GetParam() );
+
+    // fixed options
+
+    // kinematic constant penalty stiffness
+    constexpr tribol::RealT penalty = 1.0;
+    // boundary element attributes of contact surface 1, the top surface of the bottom block
+    auto contact_surf_1 = std::set<int>( { 3 } );
+    // boundary element attributes of contact surface 2, the bottom surface of the top block
+    auto contact_surf_2 = std::set<int>( { 5 } );
+
+    // create a mesh with two blocks
+    // clang-format off
+    shared::ParMeshBuilder mesh( MPI_COMM_WORLD, shared::MeshBuilder::Merged( {
+      shared::MeshBuilder::SquareMesh( 1, 1 ),
+      shared::MeshBuilder::SquareMesh( 1, 1 )
+        .translate( { 0.0, 1.0 - mesh_interpenetration } )
+        .updateAttrib( 1, 2 )
+        .updateBdrAttrib( 1, 5 )
+        .updateBdrAttrib( 3, 6 )
+      } ) );
+    // clang-format on
+
+    // grid function for higher-order nodes
+    mesh.setNodesFEColl( mfem::H1_FECollection( order, dim ) );
+
+    SetUpCommonPlaneProblem( mesh, contact_surf_1, contact_surf_2, penalty, binning_proximity, binning_method );
+  }
+
+  std::tuple<shared::ParMeshBuilder, std::set<int>, std::set<int>> SetUp3DMesh()
+  {
+    constexpr int dim = 3;
+    // polynomial order of the finite element discretization
+    auto order = std::get<0>( GetParam() );
+    // mesh interpenetration (scaled by element radius)
+    auto mesh_interpenetration = std::get<2>( GetParam() ) * std::sqrt( 2 );
+
+    // fixed options
+
+    // boundary element attributes of contact surface 1, the top surface of the bottom block
+    auto contact_surf_1 = std::set<int>( { 6 } );
+    // boundary element attributes of contact surface 2, the bottom surface of the top block
+    auto contact_surf_2 = std::set<int>( { 7 } );
+
+    // create a mesh with two cubes
+    // clang-format off
+    shared::ParMeshBuilder mesh( MPI_COMM_WORLD, shared::MeshBuilder::Merged( {
+      shared::MeshBuilder::HypercubeMesh( dim, 1 ),
+      shared::MeshBuilder::HypercubeMesh( dim, 1 )
+        .translate( { 0.0, 0.0, 1.0 - mesh_interpenetration } )
+        .updateAttrib( 1, 2 )
```

Comment:
Again, can you add comment about the different integers here. I see that we have boundary attributes 6 and 7, but not sure where 1 and 8 come from, for instance.